string place_location
string pick_location
string pick_object